File: app/views/category_color_mark.rb
class CategoryColorMark < UIView

  attr_accessor :color

  def initWithFrame(frame)
    super
    self.backgroundColor = UIColor.clearColor

    self
  end

  def initWithCoder(aDecoder)
    super
    self.backgroundColor = UIColor.clearColor

    self
  end

  def color=(color)
    @color = color

    self.setNeedsDisplay
  end

  def drawRect(rect)
    unless @color.nil?
      context = UIGraphicsGetCurrentContext()
      CGContextSetFillColorWithColor(context, @color.CGColor)
      CGContextFillEllipseInRect(context, self.bounds)
    end
  end

end

File: Forking Pasta/CategoryColorMark.swift
import UIKit

class CategoryColorMark: UIView {
    
    override init(frame: CGRect) {
        super.init(frame: frame)
        self.backgroundColor = UIColor.clearColor()
    }
    
    required init?(coder aDecoder: NSCoder) {
        super.init(coder: aDecoder)
        self.backgroundColor = UIColor.clearColor()
    }
    
    var color: UIColor? {
        didSet { setNeedsDisplay() }
    }
    
    override func drawRect(rect: CGRect) {
        if let color = color {
            let context = UIGraphicsGetCurrentContext()
            CGContextSetFillColorWithColor(context, color.CGColor)
            CGContextFillEllipseInRect(context, bounds)
        }
    }
    
}